test_host.cpp

```diff
@@ -204,13 +204,34 @@ void TestHost::DrawVertices(uint32_t elements) {
   draw_arrays(NV097_SET_BEGIN_END_OP_TRIANGLES, num_vertices);
 }
 
-void TestHost::SaveBackbuffer(const char *output_directory, const char *name) {
-  char target_file[256] = {0};
-  snprintf(target_file, 255, "%s\\%s.png", output_directory, name);
-
-  if (!CreateDirectory(output_directory, nullptr) && GetLastError() != ERROR_ALREADY_EXISTS) {
+void TestHost::EnsureFolderExists(const std::string &folder_path) {
+  if (folder_path.length() > MAX_FILE_PATH_SIZE) {
+    assert(!"Folder Path is too long.");
+  }
+  if (!CreateDirectory(folder_path.c_str(), nullptr) && GetLastError() != ERROR_ALREADY_EXISTS) {
     assert(!"Failed to create output directory.");
   }
+}
+
+// Returns the full output filepath including the filename
+// Creates output directory if it does not exist
+std::string TestHost::PrepareSaveFilePNG(std::string output_directory, const std::string &filename) {
+  std::replace(output_directory.begin(), output_directory.end(), ' ', '_');
```

Comment:
Nit: since we know we're already doing this cleanup in the constructor, we can probably omit it here. It also may be surprising behavior if the caller expected the output dir to have spaces in it for whatever reason.

Reply:
done